### PR TITLE
Build under MSVC 14

### PR DESCRIPTION
**Pull request by bkentel**
Various (incomplete) changes that allow MSVC to build the non-localized SDL tiles version.
- The strings in src/options.cpp were causing issues, so I changed some to raw-string literals.
- [CR] I couldn't test the changes I made to src/debug.cpp -- if anyone would be so kind as to compile the changes.
- [WIP] The project files still need to be cleaned up for brevity, consistency, and correctness for all build configurations, but for now "Debug TILES" builds and runs.

### CODE REVIEW

File: src/debug.cpp [hunk 1]
```diff
@@ -247,18 +247,59 @@ std::ostream &operator<<( std::ostream &out, DebugClass cl )
...
+    gettimeofday( &tv, nullptr );
+
+    auto const tt      = tv.tv_sec;
+    auto const current = localtime( &tt );
+
+    return time_info {
+        current->tm_hour
+      , current->tm_min
+      , current->tm_sec
+      , static_cast<int>(tv.tv_usec / 1000 + 0.5)
```
Comment (BevapDin) [inline src/debug.cpp]: Are you sure, this works as you intended it?
`tv_usec` is probably some kind an integer (not a floating point value), `tv.tv_usec / 1000` is an integer as well (the division discards the remainder). `tv.tv_usec / 1000 + 0.5` is now a proper floating point value, but is immediately converted back to an integer, losing the `0.5`. In other words: the static_cast is always applied to something with `.5` as fractional part.

Comment (bkentel) [inline src/debug.cpp]: I took the code as written -- looking at it again, it's a bit odd isn't it;

```
static_cast<int>(tv.tv_usec / 1000.0 + 0.5) would be more sane
```

I'll change it

**Comment by bkentel** [comment]
Updating the project files should be finished now with 696dc99.

**Comment by kevingranade** [comment]
Stuck in airports all day today and tomorrow, with luck I can look at this
Sunday.

**Comment by KA101** [comment]
Ugh.  Hoping the airports are comfy at least.

File: src/wincurse.cpp [hunk 2]
```diff
@@ -663,6 +663,10 @@ void load_colors(JsonObject &jsobj)
 #define ccolor(s) consolecolors[s][0],consolecolors[s][1],consolecolors[s][2]
 int curses_start_color(void)
 {
+    //TODO: this function strikes terror into my very soul...
```
Comment (kevingranade) [inline src/wincurse.cpp]: Snarky comments about the function are not a valid comment, please remove.

File: src/debug.cpp [hunk 3]
```diff
@@ -247,18 +247,59 @@ std::ostream &operator<<( std::ostream &out, DebugClass cl )
...
+
+        out << t.hours   << ':' << t.minutes << ':'
+            << t.seconds << '.' << t.mseconds;
+
+        return out;
+    }
+};
+
+#ifdef _MSC_VER 
+inline time_info get_time() noexcept {
```
Comment (kevingranade) [inline src/debug.cpp]: Avoid inline, the compiler has more information than you do to make that decision.

Comment (bkentel) [inline src/debug.cpp]: fair enough -- is that general policy when it isn't critical to force inline i.e. with __attribute__((always_inline))?

File: msvc140/pch.hpp [hunk 4]
```diff
@@ -0,0 +1,103 @@
+#pragma once
```
Comment (kevingranade) [inline msvc140/pch.hpp]: Normally I don't care what you're doing in the visual studio project file directory, but what the hell is this?  Why does visual studio need to shotgun include what looks like half or more of the standard library?

Comment (bkentel) [inline msvc140/pch.hpp]: precompiled headers; only (simple) way to improve build speed with MSVC.
regardless, this header is only pulled in by the MSVC build and lives outside src/ (inside msvc140/).

Comment (kevingranade) [inline msvc140/pch.hpp]: Ok, my bad, that's a totally reasonable thing to do :)

**Pull request merged**